dev-util/lockrun/lockrun-20240415.ebuild

```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs
+
+DESCRIPTION="Lockrun - runs cronjobs with overrun protection"
+HOMEPAGE="http://www.unixwiz.net/tools/lockrun.html"
+
+SRC_URI="https://downloads.uls.co.za/gentoo/lockrun/lockrun-${PV}.c.gz"
+S="${WORKDIR}"
+
+LICENSE="public-domain"
+SLOT="0"
+KEYWORDS="~amd64 ~hppa ~x86"
+
+src_prepare() {
+	default
+	mv "${P}.c" "${PN}.c" || die
+}
+
+src_compile() {
+	emake CC="$(tc-getCC)" ${PN}
+	sed '1,3 d; /^[[:space:]]*[*]\//,$ d;s/^ \*//' <"${PN}.c" > README || die
```

Comment:
This is nasty, but basically it just keeps the first comment block as the readme, if anybody has a better way to achieve this that's less nasty, please speak up.

Reply:
If upstream doesn't care about this any more, you could cut & paste it into the README (or a man page) yourself. But if he wants to keep maintaining it all inside the same one C source file... no, no ideas.

Reply:
OK, so let's stick with this mess for now, there's not been significant changes since about 2015 ... but it doesn't look like it's "unmaintained", I think it's just a very simple piece of code that doesn't require much in terms of changes, scope is well defined and it works well.  There was mention of --kill support in combination of --timeout but this never happened and is just about the only thing I can think of that may be useful, but I agree with the author that doing that *correctly* would be difficult (it's easy to send one signal, but what happens when you need to stagger signals to ensure the sub process really is dead?  Without resorting to sending SIGKILL off the bat).

Reply:
The stdin redirect (`<`) isn't really required, is it?

Reply:
I agree.  I went a bit overboard now and explained the sed logic too.  I think that'll be helpful for anyone that may need/want to modify that in the future as sed is archaic and obscure and even though I use it every day of my life I still need to reference the man page for everything but the simplest of blocks.